flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/CheckpointedInputGate.java

```diff
@@ -91,80 +83,49 @@ public CheckpointedInputGate(
 	 * checkpoint has been cancelled.
 	 *
 	 * @param inputGate The input gate to draw the buffers and events from.
-	 * @param bufferStorage The storage to hold the buffers and events for blocked channels.
 	 * @param barrierHandler Handler that controls which channels are blocked.
 	 * @param channelIndexOffset Optional offset added to channelIndex returned from the inputGate
 	 *                           before passing it to the barrierHandler.
 	 */
 	public CheckpointedInputGate(
 			InputGate inputGate,
-			BufferStorage bufferStorage,
 			CheckpointBarrierHandler barrierHandler,
 			int channelIndexOffset) {
 		this.inputGate = inputGate;
 		this.channelIndexOffset = channelIndexOffset;
-		this.bufferStorage = checkNotNull(bufferStorage);
 		this.barrierHandler = barrierHandler;
 	}
 
 	@Override
 	public CompletableFuture<?> getAvailableFuture() {
-		if (bufferStorage.isEmpty()) {
-			return inputGate.getAvailableFuture();
-		}
-		return AVAILABLE;
+		return inputGate.getAvailableFuture();
 	}
 
 	@Override
 	public Optional<BufferOrEvent> pollNext() throws Exception {
 		while (true) {
-			// process buffered BufferOrEvents before grabbing new ones
-			Optional<BufferOrEvent> next;
-			if (bufferStorage.isEmpty()) {
-				next = inputGate.pollNext();
-			}
-			else {
-				next = bufferStorage.pollNext();
-				if (!next.isPresent()) {
-					return pollNext();
-				}
-			}
+			Optional<BufferOrEvent> next = inputGate.pollNext();
 
 			if (!next.isPresent()) {
 				return handleEmptyBuffer();
 			}
 
 			BufferOrEvent bufferOrEvent = next.get();
-			if (barrierHandler.isBlocked(offsetChannelIndex(bufferOrEvent.getChannelIndex()))) {
-				// if the channel is blocked, we just store the BufferOrEvent
-				bufferStorage.add(bufferOrEvent);
-				if (bufferStorage.isFull()) {
-					barrierHandler.checkpointSizeLimitExceeded(bufferStorage.getMaxBufferedBytes());
-					bufferStorage.rollOver();
-				}
-			}
-			else if (bufferOrEvent.isBuffer()) {
+			checkState(!barrierHandler.isBlocked(offsetChannelIndex(bufferOrEvent.getChannelIndex())));
+
+			if (bufferOrEvent.isBuffer()) {
 				return next;
 			}
 			else if (bufferOrEvent.getEvent().getClass() == CheckpointBarrier.class) {
 				CheckpointBarrier checkpointBarrier = (CheckpointBarrier) bufferOrEvent.getEvent();
-				if (!endOfInputGate) {
```

Comment:
Remove of `if (!endOfInputGate)`  should be irrelevant change, but this condition is indeed redundant from my understanding. Is it right?